[CB-40] 반려동물 등록 페이지

dmstmdrbs opened the pull request:
[CB-40]

### 🔨 Jira 태스크

- CB-40
  - [하위 이슈](https://cocobob.atlassian.net/jira/software/projects/CB/boards/1?selectedIssue=CB-40) 

### 📐 구현한 내용

- 반려동물 페이지 생성
  - 디자인 반영 필요
- 프로필 사진 등록
  - image/* 형식만 받을 수 있다.
- 반려동물 종 검색 기능
  - 현재 Mock 데이터를 활용
  - API 구현되면 적용 필요
![이름입력](https://user-images.githubusercontent.com/2215762/181146505-e5d07ced-c0c8-4a6f-9be1-75344e74ee87.gif)
![나이선택](https://user-images.githubusercontent.com/2215762/181146511-30f2a448-4fd3-4071-860b-e1ab48cf6518.gif)
![종선택](https://user-images.githubusercontent.com/2215762/181146514-1173d9b4-9517-4170-bc04-9b2bc3950072.gif)


### 🚧 논의 사항

-


[CB-40]: https://cocobob.atlassian.net/browse/CB-40?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

dmstmdrbs committed:
CB-40 feat: add page for register pet

반려견 등록 페이지 추가

- 일부 페이지 구성
- IPet 타입 항목 추가
- IUser 타입 수정

dmstmdrbs committed:
CB-40 feat: add store for data of register pet

등록 중인 반려동물 정보를 위한 스토어 추가

basic페이지와 detail 페이지에서의 저장을 위한 reducer 추가

dmstmdrbs committed:
CB-134 feat: add property for onClick in FormButton

FormButton 컴포넌트에 onClick 프로퍼티 추가

dmstmdrbs committed:
CB-116 feat: add page for register pet

반려동물 등록을 위한 페이지 추가
- BasicInfo 페이지 추가
- DetailInfo 페이지 추가

추후 각 페이지에서 받을 정보는 수정될 수 있음

dmstmdrbs committed:
CB-99 refactor: change directory structure of api slice

api 모듈 디렉토리 구조 변경

apiSlice 1개로 통일
  - extended slice를 이용하여 각 endpoint별로 추가 구성

dmstmdrbs committed:
Merge branch 'develop' of https://github.com/PinkDumbbell/CocoBob-FE into CB-40-feat/addPet

dmstmdrbs committed:
CB-134 style: delete margin of text in button

dmstmdrbs committed:
Merge branch 'develop' of https://github.com/PinkDumbbell/CocoBob-FE into CB-40-feat/addPet

dmstmdrbs committed:
Merge branch 'develop' of https://github.com/PinkDumbbell/CocoBob-FE into CB-40-feat/addPet

dmstmdrbs committed:
CB-168 refator: modify redux store

redux 상태 리팩토링
- 유저 상태관리 수정
  - 페이지 전환 수정
	- 유저 로그인 상태관리 수정
- bottomSheet 상태 수정
  - 반려동물 등록에 필요한 bottomSheet 추가

dmstmdrbs committed:
CB-184 refactor: modify form input component

react-hook-form input의 에러 수정
- register 덮어쓰기 순서 수정

dmstmdrbs committed:
CB-116 refactor: modify register pet pages

반려동물 등록 페이지 컴포넌트 수정
- 각 스텝별 페이지 구성

dmstmdrbs committed:
Merge branch 'develop' of https://github.com/PinkDumbbell/CocoBob-FE into CB-40-feat/addPet

dmstmdrbs committed:
Merge branch 'CB-40-feat/addPet' of https://github.com/PinkDumbbell/CocoBob-FE into CB-40-feat/addPet

dmstmdrbs committed:
Merge branch 'CB-40-feat/addPet' of https://github.com/PinkDumbbell/CocoBob-FE into CB-40-feat/addPet

dmstmdrbs committed:
CB-186 refactor: change name of register props of FormInput

FormInput의 register props 이름 변경
- 목적에 맞는 네이밍으로 수정
- register에서 rules로 수정

dmstmdrbs committed:
CB-116 feat: modify register pet pages

반려동물 등록 페이지 구성
- redux store 수정
- 반려동물 등록 정보 타입 수정

dmstmdrbs committed:
CB-194 feat: add utility functions for date calculation

날짜 계산을 위한 dayjs 설치 및 함수 추가

dmstmdrbs committed:
CB-68 feat: add step2 page for register pet

step2 의 반려동물 나이 등록 로직 구현
- scroll picker가 없어 input[type='date']와 select + option 태그를 이용하여 구현
- redux store에 저장

dmstmdrbs committed:
CB-198 feat: add search breed bottom sheet

반려동물 검색을 위한 BottomSheet 추가

dmstmdrbs committed:
CB-199 feat: add register Pet state in blacklist of persist store

반려동물 등록 상태를 블랙리스트로 등록
- 반려동물 등록 플로우에서는 상태를 유지 할 필요가 없음

dmstmdrbs committed:
Merge branch 'CB-40-feat/addPet' of https://github.com/PinkDumbbell/CocoBob-FE into CB-40-feat/addPet

dmstmdrbs committed:
CB-181 feat: add search breed

견종 검색 로직 추가
- findBreed BottomSheet에서 검색 가능
- 추후 hook 분리 필요

dmstmdrbs committed:
CB-181 refactor: divide search breed login to custom hook

견종 검색 로직 커스텀 훅으로 분리
- useSearchBreed
- useBottomSheet

dmstmdrbs committed:
CB-200 refactor: control bottom sheet with useBottomSheet hook

bottom sheet 관리 커스텀 훅 적용

dmstmdrbs committed:
CB-201 feat: add pet enrollment api

반려동물 등록 API 연결

dmstmdrbs committed:
CB-203 refactor: change key of enrollment pet profile image

프로필 이미지 profile에서 petImage로 변경

dmstmdrbs committed:
CB-200 fix: bottom sheet unmount error

BottomSheet 컴포넌트의 unmount 구문 삭제
- bottom sheet이 아닌 bottom sheet을 사용하는 컴포넌트에서 close해야 함

dmstmdrbs committed:
CB-209 feat: add selecting image function

image를 선택하여 objectURL로 변경하는 유틸 함수 추가

dmstmdrbs committed:
CB-40 feat: add pet enrollment

반려동물 등록 기능 추가
- api 연결
- 타입 수정

dmstmdrbs committed:
CB-210 feat: validation for selected image file

선택된 파일이 이미지인지 검사하는 로직 추가

dmstmdrbs committed:
CB-201 fix: set nullable value error in pet enrollment

반려동물 등록 api에서 value가 undefined가 아니면 formData에 추가해야함

taehong0-0 commented on src/pages/RegisterPet/Step3.tsx:
```diff
@@ -0,0 +1,113 @@
+            placeholder="품종을 검색해보세요"
+          />
+          <div className="h-[50vh] px-2 py-3 overflow-y-scroll">
+            {!searchKeyword
+              ? breeds.map((breed) => (
+                  <div key={breed.breedId} onClick={() => onSelectBreed(breed)}>
+                    <p className="py-2">
+                      <span className="pr-2 text-primary-900">{breed.breedSize}</span>
+                      {breed.breedName}
+                    </p>
+                  </div>
+                ))
+              : foundBreeds.map((breed) => (
+                  <div key={breed.breedId} onClick={() => onSelectBreed(breed)}>
+                    <p className="py-2">
+                      <span className="pr-2 text-primary-900">{breed.breedSize}</span>
+                      {breed.breedName}
+                    </p>
+                  </div>
```
breeds, foundBreeds 빼고는 차이가 없는 것 같은데 합치면 좋지 않을까요??

dmstmdrbs replied:
넵 반영했습니다!

taehong0-0 reviewed (approved):
고생하셨습니다 !

dmstmdrbs committed:
CB-216 refactor: add BreedList component

견종 리스트 컴포넌트 분리
- 반복되는 부분 분리함

dmstmdrbs committed:
CB-40 refactor: change enroll flow

등록 플로우 순서 변경
- Step1. 이름
- Step2. 사진
- Step3. 견종
- Step4. 나이
- Step5. 상세 정보(성별, 몸무게, 활동레벨, 중성화, 임신/수유여부)

dmstmdrbs committed:
Merge branch 'develop' into CB-40-feat/addPet

dmstmdrbs committed:
CB-208 fix: add refresh token in header when refreshing

토큰 재발행 요청 시 refresh 토큰 헤더에 추가
